enumerables.rb

```diff
@@ -0,0 +1,67 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  def my_all?
+    output = true
+    my_each { |i| break output = false unless yield(i) }
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each { |i| break output = true unless yield(i) }
+    output
+  end
+
+  def my_none?
+    output = true
+    my_each { |i| break output = false if yield(i) }
+    output
+  end
+
+  def my_count(arg)
+    count = 0
+    my_each { |i| count += 1 if arg == i }
+    count
+  end
```

Comment:
- counts the number of elements yielding a true value if a block is given
- does not mutate the original array